### PR TITLE
Check that vtk SafeDownCast worked

### LINKED ISSUE #1244: PCLVisualizer shape handling is bad
This simple example will crash the PCLVisualizer:
```cpp
#include <pcl/io/pcd_io.h>
#include <pcl/visualization/pcl_visualizer.h>

int
main (int argc,
      char *argv[])
{
  pcl::visualization::PCLVisualizer viewer ("Viewer");
  viewer.addText("coucou", 50, 50, "text");
  pcl::PointXYZ p1 (0, 0, 0), p2 (1, 1, 1);
  viewer.addLine(p1, p2, "line");
  viewer.updateText("make the visualizer crash", 50, 50, "line");
  viewer.spin();
  return 0;
}
```

Both `text` and `line` are added in the `shape_actor_map_`, so when you try to update the ID `line` as if it was a text, the ID is found and the [`SafeDownCast`](https://github.com/PointCloudLibrary/pcl/blob/master/visualization/src/pcl_visualizer.cpp#L2666) crashes because a `vtkLODActor` (line) cannot be casted into a `vtkTextActor`:

```cpp
vtkTextActor* actor = vtkTextActor::SafeDownCast (am_it->second);
```

### PR DESCRIPTION
**Pull request by VictorLamoine**
Fixes #1244

### CODE REVIEW

File: visualization/src/pcl_visualizer.cpp [hunk 1]
```diff
@@ -892,6 +892,8 @@ bool
 pcl::visualization::PCLVisualizer::removeActorFromRenderer (const vtkSmartPointer<vtkLODActor> &actor, int viewport)
 {
   vtkLODActor* actor_to_remove = vtkLODActor::SafeDownCast (actor);
+  if (!actor_to_remove)
```
Comment (taketwo) [inline visualization/src/pcl_visualizer.cpp]: `actor` is a pointer to `vtkLODActor` by definition, I don't see how this cast may fail...

File: visualization/src/pcl_visualizer.cpp [hunk 2]
```diff
@@ -1035,6 +1041,8 @@ int
 getDefaultScalarInterpolationForDataSet (vtkDataSet* data)
 {
   vtkPolyData* polyData = vtkPolyData::SafeDownCast (data);
+  if (!polyData)
+    return (false);
```
Comment (taketwo) [inline visualization/src/pcl_visualizer.cpp]: Semantics of the return value are different in this function.

File: visualization/src/pcl_visualizer.cpp [hunk 3]
```diff
@@ -3747,6 +3782,8 @@ pcl::visualization::PCLVisualizer::renderViewTesselatedSphere (
 
      vtkSmartPointer<vtkIdTypeArray> ids = vtkSmartPointer<vtkIdTypeArray>::New ();
      ids = vtkIdTypeArray::SafeDownCast(hdw_selection->GetNode(0)->GetSelectionList());
```
Comment (taketwo) [inline visualization/src/pcl_visualizer.cpp]: Can remove `New()` on the previous line by the way.

File: visualization/src/pcl_visualizer.cpp [hunk 4]
```diff
@@ -3747,6 +3782,8 @@ pcl::visualization::PCLVisualizer::renderViewTesselatedSphere (
...
+     if (!ids)
```
Comment (taketwo) [inline visualization/src/pcl_visualizer.cpp]: Can remove `New()` two lines above by the way.

File: visualization/src/pcl_visualizer.cpp (hunk 1 above)
Comment (VictorLamoine) [inline visualization/src/pcl_visualizer.cpp]: True

File: visualization/src/pcl_visualizer.cpp (hunk 2 above)
Comment (VictorLamoine) [inline visualization/src/pcl_visualizer.cpp]: Oh right, what do I return then?

Comment (taketwo) [inline visualization/src/pcl_visualizer.cpp]: Maybe it's okay to leave it as is, with segfault possibility? It's a helper function anyway, should not be called from user code.

Comment (VictorLamoine) [inline visualization/src/pcl_visualizer.cpp]: Ok

**Comment by VictorLamoine** [comment]
Ready to be merged on my side

**Comment by taketwo** [comment]
LGTM

**Pull request merged**